Sources/SwiftWin32/Platform/Win32+Menu.swift

```diff
@@ -79,19 +79,53 @@ extension Win32MenuElement {
 }
 
 internal final class _MenuBuilder: MenuSystem {
-  internal private(set) var hMenu: MenuHandle
   internal private(set) weak var view: View?
 
   private var menus: OrderedSet<Menu>
 
   internal init?(for view: View) {
-    self.hMenu = MenuHandle(owning: CreateMenu())
-    if self.hMenu.value == nil {
+    self.view = view
+    self.menus = []
+  }
+
+  internal func build() -> MenuHandle? {
+    let rootMenu = Menu(children: Array(menus))
+    return build(rootMenu)
+  }
+
+  private func build(_ menu: Menu) -> MenuHandle? {
+    let hMenu = MenuHandle(owning: CreateMenu())
+    if hMenu.value == nil {
       log.warning("CreateMenu: \(Error(win32: GetLastError()))")
       return nil
     }
-    self.view = view
-    self.menus = []
+
+    for (index, child) in menu.children.enumerated() {
+      let imageHandle: BitmapHandle?
+      if let bitmap = child.image?.bitmap {
+        imageHandle = BitmapHandle(from: bitmap)
+      } else {
+        imageHandle = nil
+      }
+
+      var submenuHandle: MenuHandle?
+      if let submenu = child as? Menu {
+        submenuHandle = build(submenu)
+      } else {
+        submenuHandle = nil
+      }
```

Comment:
I think that this can be written more concisely.

```suggestion
      var hSubmenu: MenuHandle? =
          child is Menu ? build(child as! Menu) : nil
```

However, I think that we might be able to generalise this method a bit more and just take a `Menu?` for the parameter?

Reply:
We now have a bit more logic in the `if` block, there probably isn't a way to inline it anymore.